ssh/client.go

```diff
@@ -261,9 +260,17 @@ func NewExternalClient(sshBinaryPath, user, host string, port int, auth *Auth) (
 	// the authorized hosts file.
 	args := baseSSHArgs
 
+	// If no identities are explicitly provided, also look at the identities
+	// offered by ssh-agent
+	if len(auth.Keys) > 0 {
+		args = append(args, "-o", "IdentitiesOnly=yes")
+	}
+
 	// Specify which private keys to use to authorize the SSH request.
 	for _, privateKeyPath := range auth.Keys {
```

Comment:
Not sure what you mean here... are you talking about the case when `auth.Keys` is `0`?

I figure the normal use case will be for users to just provide the identity on the command line, which will be the same as before...